docs/adr/012_decouple_from_edge.md

```diff
@@ -52,77 +52,105 @@ The main difference comes to "ModuleConnectionHost", which is responsible for:
 
-If using Azure Kubernetes Service, a configmap to be applied to the desired cluster might work for you.
-If using Azure App Service, you might just set the environment variables in the configuration section of the application.
+As an alternative we could still use "IOTEDGE_APIVERSION" and eventually react on future changes.
 
-While these solutions are great for "static" configuration (i.e.: those values that require a restart for being changed), the LNS is coded in such a way that some configuration might be changed dynamically without having to restart the entire network server.
+#### Differentiating LNS instances in mixed-mode deployments
 
-The proposal is to **make use of "[Azure App Configuration](https://docs.microsoft.com/en-us/azure/azure-app-configuration/overview)"** to centrally manage application settings and keep [dynamic configuration using poll model](https://docs.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-aspnet-core?tabs=core5x)
+When a leaf device sends a message, its twin gets updated from the LNS with the "GatewayID" variable content, currently being set to the "IOTEDGE_DEVICEID" environment variable (defaulting to an empty string when variable is not set in the environment)
 
-### Direct method invocation
+This is the way the "Facade" Azure function has to identify and target a specific LNS instance for closing device client connections or sending C2D messages for class-c devices.
+
+The decision is to:
+
+- Prefix the **"GatewayID" with "cloud-"** for non-Edge deployments
+- Use the **"Hostname" environment variable as a default** for GatewayID field
+
+An alternative considered is to identify the difference in deployment model by querying IoT Hub for availability of a "IoT Edge" capable device and cache that result.
+
+An advantage of such alternative is the fact that there's no need to use any prefix for GatewayID which would also not tamper the observability part.
+
+A disadvantage, though, comes from the request quota limits on IoT Hub Device Registry operations which could be cause of throttling and instability for other pieces of the LNS itself.
+
+#### Direct method invocation
 
 As previously mentioned, IoT Edge is allowing the functionality of "direct method invocations" giving the possiblity to remotely invoke some functions on the LNS.
 
-When decoupling LNS from Azure IoT Edge, there are two possibilities available for handling the scenarios covered by direct method invocation.
+When decoupling LNS from Azure IoT Edge, there are different possibilities for handling the scenarios covered by direct method invocation.
+
+As stated in previous paragraphs, the "Facade" Azure Function has to differentiate between LNS deployment modes depending on the "LNS device id".
```

Comment:
don't repeat the solution here, but link to the section that discusses it. Otherwise we have this all over the place and if we do not end up using the device id, we have to change it in multiple places.